### PR TITLE
EFAX-22 added a scheduler for dealing with fax timeouts

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

https://justice.gov.bc.ca/jira/browse/EFAX-22

Added a Scheduler that polls redis for any sent faxes that have timed-out (have not received a response from exchange for the given fax).

In the [flow ](https://justice.gov.bc.ca/wiki/pages/viewpage.action?spaceKey=EFAX&title=BPEL+Process+Flow+for+eFax+Service), this is the step where the process is forced along if there is no response from exchange.

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

mvn verify

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: doc-distrib-ws/src/main/resources/application.yaml [hunk 1]
```diff
@@ -11,6 +17,14 @@ mailservice:
 
 ws:  
   faxFormat: IMCEAFAX-%RECIPIENT%+40%FAXNUMBER%@Victoria1.gov.bc.ca
+  
+  # Timeout (in milliseconds) when no reponse has been received by exhange for a sent fax
+  #   After this timeout, the fax is considered lost and a failed status is returned to the user.  
+  faxTimeout: 1500000    # 25 minutes
```
Comment (BrendanBeachBC) [inline doc-distrib-ws/src/main/resources/application.yaml:23]: Adding a comment with a conversion to a more readable time like minutes or seconds is a good idea, I like this.

File: doc-distrib-ws/src/main/resources/application.yaml [hunk 2]
```diff
@@ -11,6 +17,14 @@ mailservice:
 
 ws:  
   faxFormat: IMCEAFAX-%RECIPIENT%+40%FAXNUMBER%@Victoria1.gov.bc.ca
+  
+  # Timeout (in milliseconds) when no reponse has been received by exhange for a sent fax
```
Comment (BrendanBeachBC) [inline doc-distrib-ws/src/main/resources/application.yaml:21]: Small typo
```suggestion
  # Timeout (in milliseconds) when no response has been received by exchange for a sent fax
```

File: doc-distrib-ws/src/test/resources/application.yaml [hunk 3]
```diff
@@ -12,10 +12,19 @@ mailservice:
 spring:
   redis:
     host: ${REDIS_HOST:localhost}
+    # Port on which an embedded redis server will run for unit tests
     port: ${REDIS_PORT:6370}
 
 ws:  
   faxFormat: IMCEAFAX-%RECIPIENT%+40%FAXNUMBER%@Victoria1.gov.bc.ca
+  
+  # Timeout (in milliseconds) when no reponse has been received by exhange for a sent fax
```
Comment (BrendanBeachBC) [inline doc-distrib-ws/src/test/resources/application.yaml:21]:
```suggestion
  # Timeout (in milliseconds) when no response has been received by exchange for a sent fax
```

**Review by BrendanBeachBC — DISMISSED** [review]
Looks good to me! A couple of small typos in comments but nothing major

File: doc-distrib-ws/src/test/resources/application.yaml (hunk 3 above)
Comment (KDNeufeld) [reply doc-distrib-ws/src/test/resources/application.yaml:21]: Excellent, thanks.

**Review by BrendanBeachBC — APPROVED** [review]

**Pull request merged**